src/router/router.js

```diff
@@ -0,0 +1,46 @@
+import LoginPage from "../pages/LoginPage";
+import HomePage from "../pages/HomePage";
+import ProfilePage from "../pages/ProfilePage";
+import NotFoundPage from "../pages/NotFoundPage";
+import { Header, Footer } from "../componentes/layout";
+import { state } from "../store";
+
+export const isProd = import.meta.env.PROD;
+export const BASE_PATH = isProd ? "/front_5th_chapter1-1" : "";
+
+export const matchedPath = (path) => {
+  const currentPath = window.location.hash
+    ? window.location.hash.slice(1) || "/"
+    : window.location.pathname.replace(BASE_PATH, "") || "/";
+  return currentPath === path;
+};
+
+const root = document.getElementById("root");
+
+export const router = () => {
```

Comment:
router안에서 현재는 if-else 방식으로 모든 경로를 분기하고 있는데 다른 페이지도 만약에 추가된다면 코드가 복잡해질 수 있을 것 같아요~ 라우팅 테이블 객체를 따로 만들어서 사용해보면 어떨까요?
```suggestion
const routes = {
  "/": HomePage,
  "/login": () => state.loginState ? (navigate("/"), HomePage()) : LoginPage,
  "/profile": () => state.loginState ? ProfilePage({ Header, Footer }) : (navigate("/login"), LoginPage),
};
```